core/integration/module_config_test.go

```diff
@@ -524,6 +524,240 @@ func (ConfigSuite) TestDaggerConfig(ctx context.Context, t *testctx.T) {
 	}
 }
 
+func (ConfigSuite) TestSDKConfig(ctx context.Context, t *testctx.T) {
+	t.Run("go sdk", func(ctx context.Context, t *testctx.T) {
+		testcases := []struct {
+			name          string
+			daggerjson    string
+			expectedValue string
+			expectedError string
+		}{
+			{
+				name: "goprivate is configured",
+				daggerjson: `{
+  "name": "foo",
+  "engineVersion": "v0.16.2",
+  "sdk": {
+    "source": "go",
+    "config": {
+      "goprivate": "github.com/foobar"
+    }
+  }
+}`,
+				expectedValue: "github.com/foobar",
+			},
+			{
+				name: "invalid type for goprivate is configured",
+				daggerjson: `{
+  "name": "foo",
+  "engineVersion": "v0.16.2",
+  "sdk": {
+    "source": "go",
+    "config": {
+      "goprivate": 1234
+    }
+  }
+}`,
+				expectedError: "'GoPrivate' expected type 'string', got unconvertible type 'float64', value: '1234'",
+			},
+		}
+
+		for _, tc := range testcases {
+			t.Run(tc.name, func(ctx context.Context, t *testctx.T) {
+				c := connect(ctx, t)
+				ctr := c.Container().From(golangImage).
+					WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+					WithWorkdir("/work").
+					With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+					WithNewFile("dagger.json", tc.daggerjson).
+					WithNewFile("main.go", `package main
+
+import (
+	"os"
+)
+
+type Foo struct{}
+
+func (m *Foo) CheckEnv() string {
+	return os.Getenv("GOPRIVATE")
+}
+`)
+
+				output, err := ctr.With(daggerCall("check-env")).Stdout(ctx)
+				if tc.expectedError != "" {
+					require.NotNil(t, err)
+					execerror := err.(*dagger.ExecError)
+					require.Contains(t, execerror.Stderr, tc.expectedError)
+				} else {
+					require.Nil(t, err)
+					require.Equal(t, tc.expectedValue, output)
+				}
+			})
+		}
+	})
+
+	t.Run("module sdk", func(ctx context.Context, t *testctx.T) {
+		var withoutCustomConfigSupport = `package main
+
+import (
+	"dagger/coolsdk/internal/dagger"
+)
+
+type Coolsdk struct {
+	BarConfig string
+}
+
+func New(
+	// +default="class-default"
+	barConfig string,
+) *Coolsdk {
+	return &Coolsdk{
+		BarConfig: barConfig,
+	}
+}
+
+func (m *Coolsdk) ModuleRuntime(modSource *dagger.ModuleSource, introspectionJson *dagger.File) *dagger.Container {
+	return modSource.WithSDK("go").AsModule().Runtime().WithEnvVariable("COOL", m.BarConfig)
+}
+
+func (m *Coolsdk) Codegen(modSource *dagger.ModuleSource, introspectionJson *dagger.File) *dagger.GeneratedCode {
+	modSource = modSource.WithSDK("go")
+	return dag.GeneratedCode(
+		// apply generated diff over context directory
+		modSource.ContextDirectory().WithDirectory("/", modSource.GeneratedContextDirectory()),
+	)
+}
+`
+
+		var withCustomConfigSupport = withoutCustomConfigSupport + `
+		
+func (m *Coolsdk) WithConfig(
+	// +default="func-default"
+	barConfig string,
+) *Coolsdk {
+	m.BarConfig = barConfig
+	return m
+}
+		`
+
+		daggerjson := `{
+	"name": "foo",
+	"engineVersion": "v0.16.3-250304150120-33f537f11a03",
```

Comment:
:eyes: Should this just be `engineVersion` = `v0.16.2`? Don't think we need the dev versions here.